db/migrate/20161205142618_delete_orphaned_smart_class_parameters.rb

```diff
@@ -0,0 +1,8 @@
+class DeleteOrphanedSmartClassParameters < ActiveRecord::Migration
+  def up
+    PuppetclassLookupKey.where("NOT EXISTS (SELECT * FROM environment_classes WHERE environment_classes.puppetclass_lookup_key_id = lookup_keys.id)").destroy_all
```

Comment:
Just a comment for any onlookers, I tried `PuppetclassLookupKey.where.not(:id => EnvironmentClass.pluck(:puppetclass_lookup_key_id))` but the query produced is quite terrible (because of the .pluck), so this is probably fine.